### PR TITLE
Add stompjs types definitions.

### PR DESCRIPTION
**Pull request by JimiC**
Please fill in this template.

- [x] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.

### CODE REVIEW

**Comment by andy-ms** [comment]
Why not target the `types-2.0` branch?

**Comment by JimiC** [comment]
Not sure if what I wrote complies for targeting `types-2.0`.

**Comment by andy-ms** [comment]
It's not yet, but I'll help.

* Rebase to `types-2.0`.
* Rename `stompjs.d.ts` --> `index.d.ts`
* Remove `/// <reference path="./stompjs.d.ts" />` from the top of `stompjs-tests.ts`
* In `index.d.ts`, remove the `declare module 'stompjs' {` wrapper, and add `export` on front of `class Client`.
* Change `/// <reference path="../node/node.d.ts" />` to `/// <reference types="node" />`
* Create tsconfig.json:

```json
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6",
        "noImplicitAny": true,
        "strictNullChecks": true,
        "baseUrl": "../",
        "typeRoots": [
            "../"
        ],
        "types": [],
        "noEmit": true,
        "forceConsistentCasingInFileNames": true
    },
    "files": [
        "index.d.ts",
        "stompjs-tests.ts"
    ]
}
```

Running `tsc` in that directory should now work. (Assuming `tsc --version` is a 2.0 version.)

File: stompjs/stompjs.d.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+/// <reference path="../node/node.d.ts" />
+
+declare module 'stompjs' {
+
+    export const VERSIONS: {
+        V1_0: '1.0',
+        V1_1: '1.1',
+        V1_2: '1.2',
+        supportedVersions: () => '1.1,1.0'.split(',')
```
Comment (andy-ms) [inline stompjs/stompjs.d.ts]: I'm surprised this made it past the compiler! Definition files should not contain implementation. This appears to be equivalent to `supportedVersions(): string[]`. You could always type it as a tuple `["1.1", "1.0"]`, but I don't see why that would be useful.

File: stompjs/stompjs.d.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ack(messageID: string, subscription: string, headers?: Object): any;
+        nack(messageID: string, subscription: string, headers?: Object): any;
+    }
+
+    export interface Message {
+        command: string;
+        headers: Object;
+        body: string;
+
+        ack(headers?: Object): any;
```
Comment (andy-ms) [inline stompjs/stompjs.d.ts]: Prefer `{}` or `any` over `Object`.

File: stompjs/stompjs.d.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// Type definitions for stompjs 2.3.x
```
Comment (andy-ms) [inline stompjs/stompjs.d.ts]: Don't need the `.x`

File: stompjs/stompjs.d.ts (hunk 1 above)
Comment (JimiC) [inline stompjs/stompjs.d.ts]: Yeah. I'm trying to wrap my header about that as we speak.

File: stompjs/stompjs.d.ts (hunk 2 above)
Comment (JimiC) [inline stompjs/stompjs.d.ts]: That's something debatable. As far as I know `{}` is equivalent to `Object`. http://stackoverflow.com/questions/18961203/typescript-any-vs-object

Comment (andy-ms) [inline stompjs/stompjs.d.ts]: The [handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html) recommends against using the `Object` type. It *is* equivalent to `{}`, so as a matter of style we always use `{}` instead.

File: stompjs/stompjs.d.ts (hunk 3 above)
Comment (JimiC) [inline stompjs/stompjs.d.ts]: And how to specify that this typing is for 2.3.0+?

Comment (andy-ms) [inline stompjs/stompjs.d.ts]: `2.3` alone does that. The patch version for `@types/stompjs` will be filled in as `0` when this is published, then `1` the first time it is edited, etc.

Comment (JimiC) [inline stompjs/stompjs.d.ts]: :+1:

File: stompjs/stompjs.d.ts (hunk 2 above)
Comment (JimiC) [inline stompjs/stompjs.d.ts]: OK. I'll use `{}` then.

**Comment by JimiC** [comment]
@andy-ms Thanks a lot for helping out a first-timer. I really appreciated your help on this.
I've made a new PR (#12605) following your instructions.
As soon as that one passes all test I'm gonna close this one.

**Comment by JimiC** [comment]
Closed in favor of [types 2.0](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12605) definition.

**Pull request closed without merging**